tsl/src/chunk.c

```diff
@@ -1286,55 +1358,88 @@ typedef struct RelationSplitInfo
 	 * destination chunks.
 	 */
 	TupleConversionMap *tupmap;
-} RelationSplitInfo;
+	RowCompressor compressor;
+	RewriteStats stats;
+} RelationWriteState;
 
-static RelationSplitInfo *
-relation_split_info_create(Relation srcrel, Relation targetrel, struct VacuumCutoffs *cutoffs)
+/*
+ * SplitContext
+ *
+ * Main state for doing a split.
+ */
+typedef struct SplitContext
```

Comment:
Would it be a good idea to put all split-related things into a separate file? Seems pretty big and independent.

Reply:
Yes, it is growing big. But it is probably best to break out in a separate PR. Otherwise it will be hard to see the changes. WDYT?